src/payment/strategies/mollie/mollie-payment-strategy.ts

```diff
@@ -105,19 +118,27 @@ export default class MolliePaymentStrategy implements PaymentStrategy {
     }
 
     deinitialize(): Promise<InternalCheckoutSelectors> {
-        this._cardNumberElement?.unmount();
-        this._expiryDateElement?.unmount();
-        this._verificationCodeElement?.unmount();
-        this._cardHolderElement?.unmount();
+        this._mollieClient = undefined;
+
+        this.removeMollieComponents();
 
         return Promise.resolve(this._store.getState());
     }
 
+    private removeMollieComponents(): void {
+        const mollieComponents = document.querySelectorAll('.mollie-component');
+
+        each(mollieComponents, component => component.remove());
```

Comment:
optional, you could use the native version, `mollieComponents.forEach(component => component.remove());`